rewrite-java/src/test/java/org/openrewrite/java/internal/parser/InlineMethodCallsRecipeGeneratorTest.java

```diff
@@ -0,0 +1,166 @@
+          recipeList:
+
+            # From com.example:test-lib:1.0.0
+            - org.openrewrite.java.InlineMethodCalls:
+                methodPattern: 'com.example.TestClass method1()'
+                replacement: 'replacement1()'
+                classpathFromResources:
+                  - 'test-lib-1.0.0'
+            - org.openrewrite.java.InlineMethodCalls:
+                methodPattern: 'com.example.TestClass method2(int)'
+                replacement: 'replacement2(num)'
+                staticImports:
+                  - 'java.util.Collections.emptyList'
+                classpathFromResources:
+                  - 'test-lib-1.0.0'
+          """);
+    }
+
+    @Test
+    void skipMethodsWithoutInlineMeAnnotation(@TempDir Path tempDir) throws Exception {
+        // Create a test TypeTable TSV file with a mix of annotated and non-annotated methods
+        Path inputTsv = tempDir.resolve("test.tsv.gz");
+        String tsvContent = """
+          groupId\tartifactId\tversion\tclassAccess\tclassName\tclassSignature\tclassSuperclassSignature\tclassSuperinterfaceSignatures\taccess\tname\tdescriptor\tsignature\tparameterNames\texceptions\telementAnnotations\tparameterAnnotations\ttypeAnnotations\tconstantValue
+          com.example\ttest-lib\t1.0.0\t1\tcom/example/TestClass\t\tjava/lang/Object\t\t-1\t\t\t\t\t\t\t\t\t
+          com.example\ttest-lib\t1.0.0\t1\tcom/example/TestClass\t\tjava/lang/Object\t\t1\tregularMethod\t()V\t\t\t\t\t\t\t
+          com.example\ttest-lib\t1.0.0\t1\tcom/example/TestClass\t\tjava/lang/Object\t\t1\tannotatedMethod\t()V\t\t\t\t@Lorg/openrewrite/java/InlineMe;(replacement=s"replacement()")\t\t\t
+          """;
+        try (OutputStream os = Files.newOutputStream(inputTsv);
+             GZIPOutputStream gzos = new GZIPOutputStream(os)) {
+            gzos.write(tsvContent.getBytes());
+        }
+
+        // Run the generator
+        Path outputYaml = tempDir.resolve("output.yaml");
+        InlineMethodCallsRecipeGenerator.generate(inputTsv, outputYaml, "org.openrewrite.java.InlineMethodCallsGenerated");
+
+        // Read and verify the content
+        //language=yaml
+        assertThat(outputYaml).hasContent("""
+          #
+          # Generated InlineMe recipes from TypeTable
+          #
+
+          type: specs.openrewrite.org/v1beta/recipe
+          name: org.openrewrite.java.InlineMethodCallsGenerated
+          displayName: Inline methods annotated with `@InlineMe`
+          description: >-
+            Automatically generated recipes to inline method calls based on `@InlineMe` annotations
+            discovered in the type table.
+          recipeList:
+
+            # From com.example:test-lib:1.0.0
+            - org.openrewrite.java.InlineMethodCalls:
+                methodPattern: 'com.example.TestClass annotatedMethod()'
+                replacement: 'replacement()'
+                classpathFromResources:
+                  - 'test-lib-1.0.0'
```

Comment:
I think it would be really nice if we could change the `JavaParser.Builder#classpathFromResources()` API a bit to work with GAVs. My proposal would be that we in a first step leave the signature as is, but change it so that if the value contains a `:`, then it is interpreted as a GA or GAV (depending on if there is one or two colons). For the version part we could then also allow version ranges like `1.0.+` or `~1.0.0`

```suggestion
                  - 'com.example:test-lib:1.0.0'
```

I realize this isn't directly related to this PR.